mm2src/coins/utxo/utxo_common.rs

```diff
@@ -2087,18 +2157,72 @@ where
     }
 }
 
-pub fn check_all_utxo_inputs_signed_by_pub(
+/// This function is used to check that all inputs are signed/owned by the expected pubkey.
+///
+/// It's used to verify that all the inputs of the taker-sent dex fee are signed/owned by the taker's pubkey.
+/// It's used also by watcher to verify that all the taker payment inputs are signed/owned by the taker's pubkey.
+/// The `expected_pub` should be the taker's pubkey in compressed (33-byte) format.
+pub async fn check_all_utxo_inputs_signed_by_pub<T: UtxoCommonOps>(
+    coin: &T,
     tx: &UtxoTx,
     expected_pub: &[u8],
 ) -> Result<bool, MmError<ValidatePaymentError>> {
-    for input in &tx.inputs {
+    let expected_pub =
+        H264::from_slice(expected_pub).map_to_mm(|e| ValidatePaymentError::TxDeserializationError(e.to_string()))?;
+    let mut unsigned_tx: TransactionInputSigner = tx.clone().into();
+
+    for (idx, input) in tx.inputs.iter().enumerate() {
```

Comment:
We can do an early return here, like:

```rs
    if tx.inputs.is_empty() {
        return Ok(true);
    }
```

as `expected_pub` and `unsigned_tx` (or just `unsigned_tx` if we want to validate the `expected_pub` form) being totally waste if there is no iteration.

Reply:
a transaction is always expected to have at least one input (unless it's a coinbase and that's very unprobable). we will always perform this check but never satisfy it.

Reply:
> a transaction is always expected to have at least one input

Maybe validate that thing at the beginning of this function then?

Reply:
i think this is putting a great emphasis on a case that will never happen. but i can do it if u insist.
[a2c8f8f](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2408/commits/a2c8f8f3fb8eb94b3ecde633d797ccb9e6bfd43a)

Reply:
It's not that I insist on the first suggestion I said, what I am saying is if you expect something to always have at least one item, you usually add a sanity check on that for debugging purposes so you don't get hidden/mysterious bugs and behaviour.

Reply:
this case wouldn't be a sanity check tho. as if there are no inputs we gonna assume it's a coinbase and return `Ok(true)` (yeah we might wanna check if that's what we want at this point).

but the gist is, we always gonna assume this is a valid transaction even if it has no inputs, as this function is solely interested in verifying that: all the inputs are the taker's own inputs.

not having enough inputs to cover the total output amount, not having inputs at all, etc... is not in the interest of this func validation and is probably validates elsewhere.

Reply:
Yeah, feel free to do that.